File: Weatley.Backend/Weatley.Model/Entities/BookedRoom.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Weatley.Model.Entities
{
    public class BookedRoom : IEntityBase
    {
        public Guid Id { get; set; }
        //Relation with Room
        public Guid RoomId { get; set; }
        public Room Room { get; set; }

        //Relation with Booking
        public Guid BookingId { get; set; }
        public Booking Booking { get; set; }
    }
}

File: Weatley.Web/src/app/core/entities/bookedRoom.ts
import { Room } from './room';
import { Booking } from './booking';

export class BookedRoom {
	id: string;
	room: Room;
	roomId: string;
	booking: Booking;
	bookingId: string;

	constructor(values: Object = {}) {
		Object.assign(this, values);
	}
}
